nemo_curator/utils/split_large_files.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+
+import pyarrow as pa
+import pyarrow.parquet as pq
+import ray
+
+from nemo_curator.core.client import RayClient
+from nemo_curator.utils.file_utils import get_all_file_paths_under
+
+
+def _split_table(table: pa.Table, target_size: int) -> list[pa.Table]:
+    # Split table into two chunks
+    tables = [table.slice(0, table.num_rows // 2), table.slice(table.num_rows // 2, table.num_rows)]
+    results = []
+    for t in tables:
+        if t.nbytes > target_size:
+            # If still above the target size, continue spliting until chunks
+            # are below the target size
+            results.extend(_split_table(t, target_size=target_size))
+        else:
+            results.append(t)
+    return results
+
+
+def _write_table_to_file(table: pa.Table, outdir: str, output_prefix: str, ext: str, file_idx: int) -> int:
+    output_file = os.path.join(outdir, f"{output_prefix}_{file_idx}{ext}")
+    pq.write_table(table, output_file)
+    print(f"Saved {output_file} (~{table.nbytes / (1024 * 1024):.2f} MB)")
+    return file_idx + 1
+
+
+@ray.remote
+def split_parquet_file_by_size(input_file: str, outdir: str, target_size_mb: int) -> None:
+    root, ext = os.path.splitext(input_file)
+    if not ext:
+        ext = ".parquet"
+    outfile_prefix = os.path.basename(root)
+
+    print(f"""Splitting parquet file...
+
+Input file: {input_file}
+Output directory: {outdir}
+Target size: {target_size_mb} MB
+""")
+
```

Comment:
This is okay for now but maybe we can remove the print? 
As an FYI we mostly we use loguru (that comes with some overhead of registering loguru's serializer after ray client has started). See https://github.com/NVIDIA-NeMo/Curator/blob/b50d916312aed4ea4b38055effd74a85bffd85e0/nemo_curator/core/utils.py#L86-L90

Reply:
Sure, happy to switch over to loguru